src/data/zsqrtd/gaussian_int.lean

```diff
@@ -225,19 +225,19 @@ hp.eq_two_or_odd.elim
         from calc 1 + k * k ≤ k + k * k :
           add_le_add_right (nat.pos_of_ne_zero
             (λ hk0, by clear_aux_decl; simp [*, nat.pow_succ] at *)) _
-        ... = k * (k + 1) : by simp [add_comm, mul_add]
+        ... = k * (k + 1) : by simp [_root_.add_comm, mul_add]
```

Comment:
You can use the `@[protect_proj]` attribute on the class `principal_ideal_domain`, and then `principal_ideal_domain.add_comm` will be protected, and you won't need `_root_`

Reply:
I've just talked to Johan about this and he has suggested killing `principal_ideal_domain` completely and replacing it with a predicate `is_principal_ideal_domain` on `comm_ring`